### PR TITLE
fix(storybook): fix popper dep

### PR DESCRIPTION
**Pull request by JonasBa**
Fixes eslint dep warning for the popper hook.

https://user-images.githubusercontent.com/9317857/178502400-a524ffa0-7cf4-435f-a826-2df17a93873b.mp4

### CODE REVIEW

File: docs-ui/stories/assets/icons/popper.tsx [hunk 1]
```diff
@@ -29,31 +55,27 @@ const IconPopper = ({icon, setSelectedIcon, boxRef}: Props) => {
   const [isCircled, setIsCircled] = useState(icon.defaultProps?.isCircled ?? false);
   const [isSolid, setIsSolid] = useState(icon.defaultProps?.isSolid ?? false);
 
-  /**
-   * Generate and update code sample based on prop states
-   */
-  const getCodeSample = () => {
```
Comment (JonasBa) [inline docs-ui/stories/assets/icons/popper.tsx:35]: Didnt make sense to inline this helper to fn body so I moved it up.

File: docs-ui/stories/assets/icons/popper.tsx [hunk 2]
```diff
@@ -29,31 +55,27 @@ const IconPopper = ({icon, setSelectedIcon, boxRef}: Props) => {
...
-    return `<Icon${icon.name} color="gray500" size="${size}"${
-      isCircled ? ' isCircled' : ' '
-    }${isSolid ? ' isSolid' : ' '} />`;
-  };
-  const [codeSample, setCodeSample] = useState(getCodeSample());
```
Comment (JonasBa) [inline docs-ui/stories/assets/icons/popper.tsx:40]: the default state and setCodeSample on useEffect indicates that this should probably have been a useMemo

File: docs-ui/stories/assets/icons/popper.tsx [hunk 3]
```diff
@@ -29,31 +55,27 @@ const IconPopper = ({icon, setSelectedIcon, boxRef}: Props) => {
...
+  const codeSample = useCallback(
+    () => getCodeSample({name: icon.name, size, isCircled, isSolid}),
+    [icon.name, size, isCircled, isSolid]
+  );
 
   /**
    * Deselect icon box on outside click
    */
-  const clickAwayHandler = e => {
-    if (e.target !== boxRef && !boxRef?.contains?.(e.target)) {
```
Comment (JonasBa) [inline docs-ui/stories/assets/icons/popper.tsx:49]: This had a TS error -> property contains does not exist on RefObject as we werent actually accessing the .current

File: docs-ui/stories/assets/icons/searchPanel.tsx [hunk 4]
```diff
@@ -8,10 +8,10 @@ import space from 'sentry/styles/space';
 import {IconData, iconGroups, IconPropName, iconProps, icons} from './data';
 import IconInfoBox from './infoBox';
 
-export type ExtendedIconData = IconData & {
+export interface ExtendedIconData extends IconData {
```
Comment (malwilley) [inline docs-ui/stories/assets/icons/searchPanel.tsx:11]: Question: are we moving from `type` to `interface` and is this something I should be doing going forward?

**Review by malwilley — APPROVED** [review]

File: docs-ui/stories/assets/icons/searchPanel.tsx (hunk 4 above)
Comment (JonasBa) [reply docs-ui/stories/assets/icons/searchPanel.tsx:11]: @malwilley I would not say that we are moving from one to the other as the two are not interchangeable and each have with their own tradeoffs. I would recommend reading the tsc performance doc - I found it to be a great resource https://github.com/microsoft/TypeScript/wiki/Performance#preferring-interfaces-over-intersections when it comes to some of the internals. 

The reason why I changed is related to the type vs interface section described in the ts doc. We have historically had pretty bad compiler performance to a point where ts server would freeze on things like code save actions...

At the time I wrote an audit [doc](https://www.notion.so/sentry/Typescript-performance-audit-ac375cf3d9374e66998a6eab2d5539e4) where I summarized the changes and their impact. The original PR which was later broken down was described [here](https://github.com/getsentry/sentry/pull/30847) with followup PR's that broke it down into smaller chunks https://github.com/getsentry/sentry/pulls?q=is%3Apr+tsc+author%3AJonasBa.

Comment (malwilley) [reply docs-ui/stories/assets/icons/searchPanel.tsx:11]: Gotcha - I was aware of the performance benefit, but didn't have the other context. Thanks for the thorough explanation!

**Comment by JonasBa** [comment]
@malwilley looks like this was dropped in master, someone beat us to it :D I'm closing the PR

**Pull request closed without merging**